src/YarpUtilities/tests/YarpUtilitiesTest.cpp

```diff
@@ -0,0 +1,188 @@
+#include <random>
+
+// Catch2
+#include <catch2/catch.hpp>
+
+// YARP
+#include <yarp/os/Property.h>
+#include <yarp/os/Value.h>
+
+#include <BipedalLocomotionControllers/YarpUtilities/Helper.h>
+
+using namespace BipedalLocomotionControllers::YarpUtilities;
+
+template <typename T> void checkEqualVector(const T& vector1, const T& vector2)
+{
+    // check the size of the two vectors
+    REQUIRE(vector1.size() == vector2.size());
+
+    // iterate over all the elements
+    for (unsigned int i = 0; i < vector1.size(); i++)
+        REQUIRE(vector1[i] == vector2[i]);
+}
+
+TEST_CASE("Get int from searchable", "[int]")
+{
+    int element;
+    yarp::os::Property property;
+    std::string key = "answer_to_the_ultimate_question_of_life";
+    int value = 42;
+
+    property.put(key, value);
+
+    // test
+    REQUIRE(getElementFromSearchable(property, key, element));
+    REQUIRE(element == value);
+}
+
+TEST_CASE("Get double from searchable", "[double]")
+{
+    double element;
+    yarp::os::Property property;
+    std::string key = "pi";
+    double value = 3.14;
+
+    property.put(key, value);
+
+    // test
+    REQUIRE(getElementFromSearchable(property, key, element));
+    REQUIRE(element == value);
+}
+
+TEST_CASE("Get string from searchable", "[std::string]")
+{
+    std::string element;
+    yarp::os::Property property;
+    std::string key = "John";
+    std::string value = "Smith";
+
+    property.put(key, value);
+
+    // test
+    REQUIRE(getElementFromSearchable(property, key, element));
+    REQUIRE(element == value);
+}
+
+TEST_CASE("Get int vector from searchable", "[std::vector<int>]")
+{
+    std::vector<int> element;
+
+    yarp::os::Property property;
+    std::string key = "Fibonacci number";
+    std::vector<int> value{1, 1, 2, 3, 5, 8, 13, 21};
+    yarp::os::Value yarpValue;
+    auto list = yarpValue.asList();
+
+    for (const auto& v : value)
+        list->addInt(v);
+
+    property.put(key, yarpValue);
+
+    // test
+    REQUIRE(getVectorFromSearchable(property, key, element));
+    REQUIRE(element == value);
+}
+
+TEST_CASE("Get string vector from searchable", "[std::vector<std::string>]")
+{
+    std::vector<std::string> element;
+
+    yarp::os::Property property;
+    std::string key = "Donald's nephews";
+    std::vector<std::string> value{"Huey", "Dewey", "Louie"};
+    yarp::os::Value yarpValue;
+    auto list = yarpValue.asList();
+
+    for (const auto& v : value)
+        list->addString(v);
+
+    property.put(key, yarpValue);
+
+    // test
+    REQUIRE(getVectorFromSearchable(property, key, element));
+    REQUIRE(element == value);
+}
+
+TEST_CASE("Get bool vector from searchable", "[std::vector<bool>]")
+{
+    std::vector<bool> element;
+
+    yarp::os::Property property;
+    std::string key = "toss_coin_results";
+    std::vector<bool> value;
+    yarp::os::Value yarpValue;
+    auto list = yarpValue.asList();
+
+    // instantiate the random number generator and a Bernoulli distribution
```

Comment:
Not a big problem in this  case, but always use a fixed seed in your tests, to have deterministic tests (or perhaps `default_random_engine` has a given default seed?).

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/5/commits/3831d351a0a7f05abcaff90a57964a0c8b2815f1